src/studio/src/designer/frontend/ux-editor/containers/DesignView.tsx

```diff
@@ -1,304 +1,227 @@
-      if (updatedOrderSource.indexOf(id) > -1) {
-        // Remove component from source, place in dest layoutOrder
-        const [moveItem] = updatedOrderSource.splice(
-          updatedOrderSource.indexOf(id),
-          1,
-        );
-        updatedOrderDestination.splice(index, 0, moveItem);
-        this.setState((prevState: IDesignerPreviewState) => {
-          return {
-            ...prevState,
-            layoutOrder: {
-              ...prevState.layoutOrder,
-              [destinationContainerId]: [...updatedOrderDestination],
-              [sourceContainerId]: [...updatedOrderSource],
-            },
-            isDragging: true,
-          };
-        });
-      } else {
-        // The component has been dragged to an unknown container, locate the container and remove
-        const container = Object.keys(layoutOrder).find(
-          (containerId: string) => {
-            return layoutOrder[containerId].includes(id);
-          },
-        );
-        const [movedComponent] = layoutOrder[container].splice(
-          layoutOrder[container].indexOf(id),
-          1,
-        );
-        if (!movedComponent) {
-          return;
-        }
-        updatedOrderDestination.splice(index, 0, movedComponent);
-        this.setState((prevState: IDesignerPreviewState) => {
-          return {
-            ...prevState,
-            layoutOrder: {
-              ...prevState.layoutOrder,
-              [destinationContainerId]: [...updatedOrderDestination],
-              [container]: [...layoutOrder[container]],
-            },
-            isDragging: true,
-          };
-        });
-      }
-    }
+  const moveItemToTop = (item: EditorDndItem) => {
+    const arr = state.layoutOrder[item.containerId];
+    swapItemsInsideTheSameContainer(item, arr[0]);
   };
 
-  public getStatefullIndexOfContainer = (
-    containerId: string,
-    parentContainerId: string = Object.keys(this.props.layoutOrder)[0],
-  ): number => {
-    if (containerId === parentContainerId) {
-      return 0;
-    }
-    return this.state.layoutOrder[parentContainerId]?.indexOf(containerId);
+  const moveItemToBottom = (item: EditorDndItem) => {
+    const arr = state.layoutOrder[item.containerId];
+    swapItemsInsideTheSameContainer(item, arr[arr.length - 1]);
   };
 
-  public moveContainer = (
-    id: string,
-    index: number,
-    sourceContainerId: string,
-    destinationContainerId: string,
-  ) => {
-    if (!id) {
-      // No id, no drag
-      return;
+  const swapItemsInsideTheSameContainer = (
+    movedItem: EditorDndItem,
+    targetId: string,
+  ): void => {
+    const currentLayoutOrder = state.layoutOrder[movedItem.containerId];
+    const newLayoutOrder = swapArrayElements(
+      currentLayoutOrder,
+      movedItem.id,
+      targetId,
+    );
+    setContainerLayoutOrder(movedItem.containerId, newLayoutOrder);
+    movedItem.index = newLayoutOrder.indexOf(movedItem.id);
+  };
+
+  const moveItem = (
```

Comment:
Would suggest to remove the comments.

If you dont think the conditionals are explanatory enough, I would suggest to extract them to separate variables with meaningful names, fex:

```tsx
//instead of this:
if (movedItem.id === targetItem.id) {
  return; // we are hovering our selves... no need to do anything.
}

const isHoveringSelf = movedItem.id === targetItem.id;
if (isHoveringSelf) {
  return; 
}


```